Automattic/wp-calypso
# Components: Consolidate PostSelector styles for standalone usage

Prerequisite for #303

This pull request seeks to migrate many of the styles applied to the `<PostSelector />` component in the context of the page editor sidebar to more generalized usage. It also adds the component to the [DevDocs App Components](http://calypso.localhost:3000/devdocs/app-components) page as an example [app component](https://github.com/Automattic/wp-calypso/blob/master/docs/our-approach-to-data.md#app-components).

![App Component](https://cloud.githubusercontent.com/assets/1779930/12929095/ff09d482-cf3e-11e5-9540-aa8d6db63bda.png)

__Implementation notes:__

These revisions also fix an issue where post titles shown in the selector would appear as encoded, e.g. `Ribs &amp; Chicken` instead of `Ribs & Chicken`.

__Design notes:__

These revisions also change the search field to be fixed to the top of the non-compact post selector. This is arguably a better user experience, since the search should always be available to the user regardless of search position.

__Testing instructions:__

Verify that the existing `<PostSelector />` used in the [Calypso page editor](http://calypso.localhost:3000/page) remains unaffected, aside from the single design note described above.

Also verify that the [DevDocs App Components](http://calypso.localhost:3000/devdocs/app-components) page includes a functioning example component that displays well as a standalone component.
